pingora-core/src/server/mod.rs

```diff
@@ -168,6 +168,30 @@ impl Server {
         Ok(())
     }
 
+    /// Create a new [`Server`], using the [`Opt`] and [`ServerConf`] values provided
+    ///
+    /// This method is intended for pingora frontends that are NOT using the built-in
+    /// command line and configuration file parsing, and are instead using their own.
+    ///
+    /// If a configuration file path is provided as part of `opt`, it will be ignored
+    /// and a warning will be logged.
+    pub fn new_with_opt_and_conf(opt: Opt, conf: ServerConf) -> Server {
+        if let Some(c) = opt.conf.as_ref() {
+            warn!("Ignoring command line argument using '{c}' as configuration, and using provided configuration instead.");
+        }
+        let (tx, rx) = watch::channel(false);
+
+        Server {
+            services: vec![],
+            listen_fds: None,
+            shutdown_watch: tx,
+            shutdown_recv: rx,
+            configuration: Arc::new(conf),
```

Comment:
I think it makes sense to merge relevant `Opt` and `ServerConf` fields like we do here, so that it's consistent with the other constructor functions: https://github.com/cloudflare/pingora/pull/165/files#diff-fce3c0bab46ec8ecc543fce29971e783c65cae72df5d44807b0407de9bddfe62R158-R160

We can probably extract that into a `ServerConf::merge_opt()` function.

Reply:
(Right now practically speaking this would mean the `daemon` server conf option may be overridden when `Opt` wants daemonization.)